src/main/java/org/voltdb/exportclient/KinesisFirehoseExportClient.java

```diff
@@ -0,0 +1,264 @@
+        String secretKey = config.getProperty("secret.key","").trim();
+        if (secretKey.isEmpty()) {
+            throw new IllegalArgumentException("KinesisFirehoseExportClient: must provide a secret.key");
+        }
+        m_firehoseClient = new AmazonKinesisFirehoseClient(
+                new BasicAWSCredentials(accessKey, secretKey));
+        m_firehoseClient.setRegion(region);
+
+        m_skipInternals = Boolean.parseBoolean(config.getProperty("skipinternals", "false"));
+
+        final TimeZone tz = TimeZone.getTimeZone(config.getProperty("timezone", VoltDB.GMT_TIMEZONE.getID()));
+        m_ODBCDateformat = new ThreadLocal<SimpleDateFormat>() {
+            @Override
+            protected SimpleDateFormat initialValue() {
+                SimpleDateFormat sdf = new SimpleDateFormat(Constants.ODBC_DATE_FORMAT_STRING);
+                sdf.setTimeZone(tz);
+                return sdf;
+            }
+        };
+
+        m_binaryEncoding = ExportDecoderBase.BinaryEncoding.valueOf(
+                config.getProperty("binaryencoding", "HEX").trim().toUpperCase());
+
+        // Default to CSV if missing
+        String type = config.getProperty("type", "csv").trim();
+        if (type.equalsIgnoreCase("csv")) {
+            m_seperator = ',';
+        } else if (type.equalsIgnoreCase("tsv")) {
+            m_seperator = '\t';
+        } else {
+            throw new IllegalArgumentException("Error: type must be one of CSV or TSV");
+        }
+
+        validateStream();
+        // Kinesis firehose limits record size to 1,000 KB
+        setRowLengthLimit(1000000, m_skipInternals);
+    }
+
+    private void validateStream() throws Exception {
+        DescribeDeliveryStreamRequest describeHoseRequest = new DescribeDeliveryStreamRequest().
+                withDeliveryStreamName(m_streamName);
+        DescribeDeliveryStreamResult  describeHoseResult = null;
+        String status = "UNDEFINED";
+        describeHoseResult = m_firehoseClient.describeDeliveryStream(describeHoseRequest);
+        status = describeHoseResult.getDeliveryStreamDescription().getDeliveryStreamStatus();
+        if(status.equalsIgnoreCase("ACTIVE")){
+            return;
+        }
+        else if(status.equalsIgnoreCase("CREATING")){
+            Thread.sleep(5000);
+            validateStream();
+        }
+        else {
+            throw new Exception("Cannot use stream " + m_streamName + ", responded with " + status);
+        }
+    }
+
+    @Override
+    public void shutdown()
+    {
+        m_firehoseClient.shutdown();
+    }
+
+    @Override
+    public ExportDecoderBase constructExportDecoder(AdvertisedDataSource source)
+    {
+        return new KinesisFirehoseExportDecoder(source);
+    }
+
+    class KinesisFirehoseExportDecoder extends ExportDecoderBase {
+        private final ListeningExecutorService m_es;
+
+        private ArrayList<Record> m_records;
+
+        @Override
+        public ListeningExecutorService getExecutor() {
```

Comment:
you need one for each decoder to speed things up. see other export implementations. i.e. KafkaExportClient